src/commands.rs

```diff
@@ -100,94 +100,92 @@ fn get_port() -> u16 {
 /// Re-execute the current executable as a background server, and wait
 /// for it to start up.
 #[cfg(not(windows))]
-fn run_server_process() -> io::Result<ServerStartup> {
+fn run_server_process() -> Result<ServerStartup> {
     use libc::{c_int, poll, pollfd, nfds_t, POLLIN, POLLERR};
     use tempdir::TempDir;
     use std::os::unix::io::AsRawFd;
     use std::os::unix::net::UnixListener;
 
     trace!("run_server_process");
-    let tempdir = try!(TempDir::new("sccache"));
+    let tempdir = TempDir::new("sccache")?;
     let socket_path = tempdir.path().join("sock");
-    let listener = try!(UnixListener::bind(&socket_path));
-    env::current_exe()
-        .and_then(|exe_path| {
-            process::Command::new(exe_path)
-                .env("SCCACHE_START_SERVER", "1")
-                .env("SCCACHE_STARTUP_NOTIFY", &socket_path)
-                .env("RUST_BACKTRACE", "1")
-                .spawn()
-        })
-        .and_then(|_| {
-            // wait for a connection on the listener using `poll`
-            let mut pfds = vec![pollfd {
-                fd: listener.as_raw_fd(),
-                events: POLLIN | POLLERR,
-                revents: 0,
-            }];
-            loop {
-                match unsafe { poll(pfds.as_mut_ptr(), pfds.len() as nfds_t, SERVER_STARTUP_TIMEOUT_MS as c_int) } {
-                    // Timed out.
-                    0 => return Ok(ServerStartup::TimedOut),
-                    // Error.
-                    -1 => {
-                        let e = Error::last_os_error();
-                        match e.kind() {
-                            // We should retry on EINTR.
-                            ErrorKind::Interrupted => {}
-                            _ => return Err(e),
-                        }
-                    }
-                    // Success.
-                    _ => {
-                        if pfds[0].revents & POLLIN == POLLIN {
-                            // Ready to read
-                            break;
-                        }
-                        if pfds[0].revents & POLLERR == POLLERR {
-                            // Could give a better error here, I suppose?
-                            return Ok(ServerStartup::TimedOut);
-                        }
-                    }
+    let listener = UnixListener::bind(&socket_path)?;
```

Comment:
I guess we could replace most of this with tokio-uds, right? (It'd be nice to get rid of some unsafe code.)